snap/snapcraft.yaml

```diff
@@ -0,0 +1,171 @@
+name: netdata # you probably want to 'snapcraft register <name>'
+base: core24 # the base snap is the execution environment for this snap
+build-base: devel
+version: '0.1' # just for humans, typically '1.2+git' or '1.3.2'
+summary: Netdata Agent
+description: |
+  Netdata is distributed, real-time, performance and health monitoring for
+  systems and applications. It provides insights of everything happening on the
+  systems it runs using interactive web dashboards.
+
+grade: devel # must be 'stable' to release into candidate/stable channels
+confinement: strict # use 'strict' once you have the right plugs and slots
+
+layout:
+#  /etc/netdata:
+#    symlink: $SNAP_COMMON/etc/netdata
+#  /usr/lib/netdata:
+#    symlink: $SNAP/usr/lib/netdata
+#  /usr/libexec/netdata:
+#    symlink: $SNAP/usr/libexec/netdata
+#  /usr/share/netdata:
+#    symlink: $SNAP/usr/share/netdata
+#  /var/cache/netdata:
+#    symlink: $SNAP_DATA/var/cache/netdata
+#  /var/lib/netdata:
+#    symlink: $SNAP_DATA/var/lib/netdata
+#  /var/log/netdata:
+#    symlink: $SNAP_DATA/var/log/netdata
+  /etc/sensors3.conf:
+    symlink: $SNAP/etc/sensors3.conf
+  /usr/bin/lsns:
+    bind-file: $SNAP/usr/bin/lsns
+
+parts:
+  deps:
+    plugin: nil
+    source: .
+    source-type: local
+    override-build: |
+      packaging/installer/install-required-packages.sh --dont-wait --non-interactive -i netdata-all
+
+
+  netdata:
+    plugin: cmake
+    source: .
+    source-type: local
+    build-environment:
+      - CFLAGS: ${CFLAGS:-"-O2 pipe"}
+      - DESTDIR: $CRAFT_PART_INSTALL
+    build-packages:
+      - golang-1.21-go
+      - libbrotli-dev
+      - libcap-dev
+      - libcups2-dev
+      - libipmimonitoring-dev
+      - libmongoc-dev
+      - libnetfilter-acct-dev
+      - libpcre2-dev
+      - libprotobuf-dev
+      - libsnappy-dev
+      - libxen-dev
+      - protobuf-compiler
+    cmake-generator: Ninja
+    cmake-parameters:
+      - -DCMAKE_INSTALL_PREFIX=/snap/netdata/current
+      - -DENABLE_ACLK=On
+      - -DENABLE_CLOUD=On
+      - -DENABLE_DBENGINE=On
+      - -DENABLE_H2O=On
+      - -DENABLE_ML=On
+      - -DENABLE_PLUGIN_APPS=On
+      - -DENABLE_PLUGIN_CGROUP_NETWORK=On
+      - -DENABLE_PLUGIN_DEBUGFS=On
+      - -DENABLE_PLUGIN_EBPF=On
+      - -DENABLE_PLUGIN_FREEIPMI=On
+      - -DENABLE_PLUGIN_GO=On
+      - -DENABLE_PLUGIN_LOCAL_LISTENERS=On
+      - -DENABLE_PLUGIN_LOGS_MANAGEMENT=On
+      - -DENABLE_PLUGIN_NFACCT=On
+      - -DENABLE_PLUGIN_PERF=On
+      - -DENABLE_PLUGIN_SLABINFO=On
+      - -DENABLE_PLUGIN_SYSTEMD_JOURNAL=On
+      - -DENABLE_PLUGIN_XENSTAT=On
```

Comment:
I don’t _think_ it’s possible for a Snap package to access the required interfaces for this plugin to work, no matter what connections are made, so we should probably just disable it.

Reply:
You mean Xen specifically? Yeah, I haven't gotten to testing that and will indeed disable it if that is indeed the case.

Fortunately, many things _do_ work. I'm currently trying to get eBPF working properly, but here too, some interfaces don't appear to be exposed and may require putting in a ticket with snapd.